tests/impls/gen_object_detector_blackbox_sal/test_occlusion_based.py

```diff
@@ -14,6 +14,31 @@
 
 class TestPerturbationOcclusion:
 
+    class StubPI (PerturbImage):
+        """
+        Stub perturber that returns masks of ones.
+        """
+
+        def perturb(self, ref_image: np.ndarray) -> np.ndarray:
+            return np.ones((6, *ref_image.shape[:2]), dtype=bool)
+
+        get_config = None  # type: ignore
+
+    class StubGen (GenerateDetectorProposalSaliency):
+        """
+        Stub saliency generator that returns zeros with correct shape.
+        """
+
+        def generate(
+            self,
+            ref_dets: np.ndarray,
+            pert_dets: np.ndarray,
+            pert_masks: np.ndarray
+        ) -> np.ndarray:
+            return np.zeros((ref_dets.shape[0], *pert_masks.shape[1:]), dtype=np.float16)
+
+        get_config = None  # type: ignore
```

Comment:
Looks like these were here before, but now that we're adding even more trivial impls, does it make sense to use mocking instead? (Applies to other test impls too)